message/v4/synchronous_message_test.go

```diff
@@ -78,65 +67,125 @@ func TestSyncTypeSystem(t *testing.T) {
-			Plugin:  "protobuf",
-			Version: "0.3.15",
-		}).
-		WithContents(grpcInteraction, "application/protobuf").
-		StartTransport("grpc", "127.0.0.1", nil). // For plugin tests, we can't assume if a transport is needed, so this is optional
-		ExecuteTest(t, func(t TransportConfig, m SynchronousMessage) error {
-			fmt.Println("Executing a test - this is where you would normally make the gRPC call")
-
-			return nil
+		err := p.AddSynchronousMessage("some description").
+			Given("some state").
+			UsingPlugin(PluginConfig{
+				Plugin:  "csv",
+				Version: "0.0.6",
+			}).
+			WithContents(csvInteraction, "text/csv").
+			ExecuteTest(t, func(m SynchronousMessage) error {
+				fmt.Println("Executing the CSV test")
+				return nil
+			})
+
+		assert.NoError(t, err)
+
+	}
+}
+func TestSyncTypeSystem_ProtobufPlugin_Matcher_Transport(t *testing.T) {
+	if os.Getenv("SKIP_PLUGIN_PROTOBUF") != "1" {
+		_ = log.SetLogLevel("INFO")
+		p, _ := NewSynchronousPact(Config{
+			Consumer: "consumer",
+			Provider: "provider",
+			PactDir:  "/tmp/",
 		})
+		dir, _ := os.Getwd()
+		path := fmt.Sprintf("%s/../../internal/native/pact_plugin.proto", strings.ReplaceAll(dir, "\\", "/"))
 
-	assert.Error(t, err)
-
-	p, _ = NewSynchronousPact(Config{
-		Consumer: "consumer",
-		Provider: "provider",
-		PactDir:  "/tmp/",
-	})
+		grpcInteraction := `{
+		"pact:proto": "` + path + `",
+		"pact:proto-service": "PactPlugin/InitPlugin",
+		"pact:content-type": "application/protobuf",
+		"request": {
+			"implementation": "notEmpty('pact-go-driver')",
+			"version": "matching(semver, '0.0.0')"	
+		},
+		"response": {
+			"catalogue": [
+				{
+					"type": "INTERACTION",
+					"key": "test"
+				}
+			]
+		}
+	}`
 
-	// Sync - with plugin + transport (fail)
-	err = p.AddSynchronousMessage("some description").
-		Given("some state").
-		UsingPlugin(PluginConfig{
-			Plugin:  "protobuf",
-			Version: "0.3.15",
-		}).
-		WithContents(grpcInteraction, "application/protobuf").
-		StartTransport("grpc", "127.0.0.1", nil). // For plugin tests, we can't assume if a transport is needed, so this is optional
-		ExecuteTest(t, func(t TransportConfig, m SynchronousMessage) error {
-			fmt.Println("Executing a test - this is where you would normally make the gRPC call")
+		// Sync - with plugin + transport (pass)
+		err := p.AddSynchronousMessage("some description").
+			Given("some state").
+			UsingPlugin(PluginConfig{
+				Plugin:  "protobuf",
+				Version: "0.3.15",
+			}).
+			WithContents(grpcInteraction, "application/protobuf").
+			StartTransport("grpc", "127.0.0.1", nil). // For plugin tests, we can't assume if a transport is needed, so this is optional
+			ExecuteTest(t, func(t TransportConfig, m SynchronousMessage) error {
+				fmt.Println("Executing a test - this is where you would normally make the gRPC call")
+
+				return nil
+			})
+
+		assert.Error(t, err)
+		// assert.Equal(t, "Did not receive any requests for path 'PactPlugin/InitPlugin'", err)
```

Comment:
This test is suspect. It previously didn't have an assertion and the error message isn't helpful to users

One would expect to see `Did not receive any requests for path 'PactPlugin/InitPlugin'`

```
		Error
		synchronous_message_test.go:130:
		Error Trace:    /Users/saf/dev/pact-foundation/pact-go/message/v4/synchronous_message_test.go:130
		Error:          Not equal:
						expected: string("Did not receive any requests for path 'PactPlugin/InitPlugin'")
						actual  : *errors.errorString(&errors.errorString{s:"pact validation failed: [{Request:{Method: Path:PactPlugin/InitPlugin Query: Headers:map[] Body:<nil>} Mismatches:[] Type:}]"})
						
```

The logs show a sane error, which I've multi-lined for readability

```
2024-07-04T01:36:33.745740Z DEBUG ThreadId(01) pact_plugin_driver::plugin_manager: 
Got response: ShutdownMockServerResponse 
{ ok: false, 
results: [
MockServerResult 
{ 
path: "PactPlugin/InitPlugin", 
error: "Did not receive any requests for path 'PactPlugin/InitPlugin'", 
mismatches: [] }
] }
```